tests/unit/project_loader/test_config.py

```diff
@@ -1574,6 +1576,130 @@ def test_invalid_yaml_invalid_epochs(self):
+            'expected_armhf': ['armhf'],
+            'yaml': '[amd64, i386]',
+        }),
+        ('single object list', {
+            'expected_amd64': ['amd64'],
+            'expected_i386': ['i386'],
+            'expected_armhf': ['armhf'],
+            'yaml': dedent("""
+                - build-on: [amd64]
+                  run-on: [amd64]
+            """),
+        }),
+        ('multiple object list', {
+            'expected_amd64': ['amd64'],
+            'expected_i386': ['i386'],
+            'expected_armhf': ['armhf', 'arm64'],
+            'yaml': dedent("""
+                - build-on: [amd64]
+                  run-on: [amd64]
+                - build-on: [i386]
+                  run-on: [i386]
+                - build-on: [armhf]
+                  run-on: [armhf, arm64]
+            """),
+        }),
+        ('omit run-on', {
+            'expected_amd64': ['amd64'],
+            'expected_i386': ['i386'],
+            'expected_armhf': ['armhf'],
+            'yaml': dedent("""
+                - build-on: [amd64]
+            """),
+        }),
+        ('single build-on string, no list', {
+            'expected_amd64': ['amd64'],
+            'expected_i386': ['i386'],
+            'expected_armhf': ['armhf'],
+            'yaml': dedent("""
+                - build-on: amd64
+            """),
+        }),
+        ('build- and run-on string, no lists', {
+            'expected_amd64': ['amd64'],
+            'expected_i386': ['amd64'],
+            'expected_armhf': ['armhf'],
+            'yaml': dedent("""
+                - build-on: i386
+                  run-on: amd64
+            """),
+        }),
+        ('build on all', {
+            'expected_amd64': ['amd64'],
+            'expected_i386': ['amd64'],
+            'expected_armhf': ['amd64'],
+            'yaml': dedent("""
+                - build-on: [all]
+                  run-on: [amd64]
+            """),
+        }),
+        ('run on all', {
+            'expected_amd64': ['all'],
+            'expected_i386': ['i386'],
+            'expected_armhf': ['armhf'],
+            'yaml': dedent("""
+                - build-on: [amd64]
+                  run-on: [all]
+            """),
+        }),
+    ]
+
+    arch_scenarios = [
+        ('amd64', {'arch': 'amd64'}),
+        ('i386', {'arch': 'i386'}),
+        ('armhf', {'arch': 'armhf'}),
+    ]
+
+    scenarios = multiply_scenarios(yaml_scenarios, arch_scenarios)
+
+    def test_architectures(self):
+        snippet = ''
+        if self.yaml:
+            snippet = 'architectures: {}'.format(self.yaml)
+        self.make_snapcraft_yaml(dedent("""\
+            name: test
+            version: "1"
+            summary: test
+            description: test
+            {}
+            parts:
+              my-part:
+                plugin: nil
+        """).format(snippet))
+
+        # with open(os.path.join('snap', 'snapcraft.yaml'), 'r') as f:
```

Comment:
rogue comment

Reply:
Oh whoops, I knew that was going to bite me when I commented it out :grin: .

Reply:
Removed.